packages/core/src/lib/wallet/wallet.types.ts

```diff
@@ -80,7 +80,7 @@ interface BaseWalletBehaviour {
   signAndSendTransactions(
     params: SignAndSendTransactionsParams
   ): Promise<Array<providers.FinalExecutionOutcome>>;
-  signMessage(params: SignMessageParams): Promise<SignedMessage | void>;
+  signMessage?(params: SignMessageParams): Promise<SignedMessage | void>;
```

Comment:
Why is this method optional? It is now defined by the decorator, no?

It may throw an error, but its not optional. Dapps should just assume its defined but may throw an error,.

Reply:
It is made optional here so typescript does not throw/complain if the signMessage is not present at a specific wallet.

The `signMessage` will be actually present in all modules when trying to call it from the dApp side, it's just that if it's not present in the specific wallet it will throw a consistent/same error for all wallets.

Once you get the wallet/module:
```ts
const meteor = await selector.wallet("meteor-wallet"); 
//At this point the module has been decorated and `signMessage` exists when calling it:

await meteor.signInMessage({
    message: "test message",
    recipient: "www.google.com",
    nonce: [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,24,26,27,28,29,30,31],
});
// If it's implemented in the wallet it returns the signedMessage, if not it throws the error.
```

Reply:
Okay cool- so at this point:

```ts
const meteor = await selector.wallet("meteor-wallet"); 
```

`meteor` is no longer an optional value? It is defined, and won't give me type errors forcing me to run checks before using it? (such as `if (meteor.signMessage !== undefined) { ... }`)

Its more about the Types. I'm not sure if the decorator will change the value from optional to defined. But if it does at that point, its all good.

Reply:
Hi, you are right in this PR the decorator will not add/set the `signMessage` as defined I was working on a fix before replying here so I will add the fix in a follow up PR: https://github.com/Elabar/wallet-selector/pull/2 .

Sorry for the late response as I was trying to add this fix.